xio-core/src/main/java/com/xjeffrose/xio/http/ProxyBackendHandler.java

```diff
@@ -54,7 +55,7 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
     // TODO(CK): this should really be some sort of notification to the frontend
     // that the backend closed. Keepalive/h2 will require the connection to stay open, we
     // shouldn't be closing it.
-    frontend.close();
+    frontend.fireUserEventTriggered(ProxyClientIdle.INSTANCE);
```

Comment:
This isn’t idle so much as closed. 

https://netty.io/4.1/api/io/netty/channel/ChannelInboundHandler.html#channelInactive-io.netty.channel.ChannelHandlerContext-